vault/core_metrics_test.go

```diff
@@ -140,6 +140,63 @@ func TestCoreMetrics_KvSecretGauge(t *testing.T) {
 	}
 }
 
+func TestCoreMetrics_KvSecretGauge_BadPath(t *testing.T) {
+	// Use the real KV implementation instead of Passthrough
+	AddTestLogicalBackend("kv", logicalKv.Factory)
+	// Clean up for the next test.
+	defer func() {
+		delete(testLogicalBackends, "kv")
+	}()
+	core, _, _ := TestCoreUnsealed(t)
+
+	me := &MountEntry{
+		Table:   mountTableType,
+		Path:    sanitizePath("kv1"),
+		Type:    "kv",
+		Options: map[string]string{"version": "1"},
+	}
+	ctx := namespace.RootContext(nil)
+	err := core.mount(ctx, me)
+	if err != nil {
+		t.Fatalf("mount error: %v", err)
+	}
+
+	// I don't think there's any remaining way to create a zero-length
+	// key via the API, so we'll fake it by talking to the storage layer directly.
+	fake_entry := &logical.StorageEntry{
```

Comment:
how is using this generating a zero-length key?

Reply:
The key in storage is /logical/<uuid assigned to secret engine>/foo/

If you then do a list of kv1/foo you'll get the key "" because of the trailing slash, which is interpreted as a directory marker.